### PR TITLE
CustomGlobalExceptionHandler

### PR DESCRIPTION
**Pull request by VKhmil**
Classes and interfaces added
-CustomGlobalExceptionHandler
-IsbnValidator
-Isbn (annotation)

Used annotation
-Valid
-Isbn (custom)
-NotNull
-Min

### CODE REVIEW

File: src/main/java/com/bookappstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
```
Comment (Rommelua) [inline src/main/java/com/bookappstore/dto/CreateBookRequestDto.java]: use NotBlank

Comment (VKhmil) [reply src/main/java/com/bookappstore/dto/CreateBookRequestDto.java]: thanks, much better

File: src/main/java/com/bookappstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    private String title;
+    @NotBlank
+    private String author;
+    @NotBlank
+    @Isbn
+    private String isbn;
+    @NotNull
+    @Min(0)
+    private BigDecimal price;
+    private String description;
```
Comment (MentorMA) [inline src/main/java/com/bookappstore/dto/CreateBookRequestDto.java:25]: You can validate description too

File: src/main/java/com/bookappstore/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    @ExceptionHandler(value = EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException ex,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NOT_FOUND);
+        body.put("error", "Entity not found: " + ex.getMessage());
```
Comment (MentorMA) [inline src/main/java/com/bookappstore/exception/CustomGlobalExceptionHandler.java:56]: Let save strings to constants

File: src/main/java/com/bookappstore/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import com.bookappstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
+
```
Comment (MentorMA) [inline src/main/java/com/bookappstore/mapper/BookMapper.java]:
```suggestion
```

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookappstore/mapper/BookMapper.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import com.bookappstore.config.MapperConfig;
+import com.bookappstore.dto.BookDto;
+import com.bookappstore.dto.CreateBookRequestDto;
+import com.bookappstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Comment (okuzan) [inline src/main/java/com/bookappstore/mapper/BookMapper.java:11]:
```suggestion
```

File: src/main/java/com/bookappstore/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (okuzan) [inline src/main/java/com/bookappstore/exception/CustomGlobalExceptionHandler.java:21]:
```suggestion
```

File: src/main/java/com/bookappstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (okuzan) [reply src/main/java/com/bookappstore/dto/CreateBookRequestDto.java:25]: not blank + size would be enough, no need for a custom validator I think

File: src/main/java/com/bookappstore/validation/Isbn.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package com.bookappstore.validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
```
Comment (okuzan) [inline src/main/java/com/bookappstore/validation/Isbn.java:10]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**